amgut/lib/data_access/ag_data_access.py

```diff
@@ -230,172 +198,154 @@ def registerHandoutKit(self, ag_login_id, supplied_kit_id):
-            cur.execute("""SELECT agc.participant_name,
-                                  agc.participant_email,
-                                  agc.parent_1_name,
-                                  agc.parent_2_name,
-                                  agc.is_juvenile,
-                                  agc.deceased_parent,
-                                  agc.ag_login_id,
-                                  agc.date_signed,
-                                  agc.assent_obtainer,
-                                  agc.age_range,
-                                  agl.survey_id
-                           FROM ag_consent agc JOIN
-                                ag_login_surveys agl
-                                USING (ag_login_id, participant_name)
-                           WHERE agl.survey_id=%s""", [survey_id])
+        with TRN:
+            TRN.add("""SELECT agc.participant_name,
+                              agc.participant_email,
+                              agc.parent_1_name,
+                              agc.parent_2_name,
+                              agc.is_juvenile,
+                              agc.deceased_parent,
+                              agc.ag_login_id,
+                              agc.date_signed,
+                              agc.assent_obtainer,
+                              agc.age_range,
+                              agl.survey_id
+                           FROM ag_consent agc
+                           JOIN ag_login_surveys agl
+                           USING (ag_login_id, participant_name)
+                           WHERE agl.survey_id = %s""", [survey_id])
             # Can refactor this with _get_col_names_from_cursor()
-            colnames = [x[0] for x in cur.description]
-            result = cur.fetchone()
-            if result:
-                result = {k: v for k, v in zip(colnames, result)}
-                if 'date_signed' in result:
-                    result['date_signed'] = str(result['date_signed'])
-                return result
+            result = TRN.execute_fetchindex()
+            if not result:
+                raise ValueError("Survey ID does not exist in DB: %s" %
+                                 survey_id)
+            return dict(result[0])
 
     def logParticipantSample(self, ag_login_id, barcode, sample_site,
                              environment_sampled, sample_date, sample_time,
                              participant_name, notes):
-
-        conn_handler = SQLConnectionHandler()
-        if sample_site is not None:
-            # Get survey id
-            sql = """SELECT survey_id
-                     FROM ag_login_surveys
-                     WHERE ag_login_id = %s AND participant_name = %s"""
-
-            survey_id = conn_handler.execute_fetchone(
-                sql, (ag_login_id, participant_name))
-            if survey_id:
-                # remove the list encapulation
-                survey_id = survey_id[0]
+        with TRN:
```

Comment:
My understanding of the idea of the Transaction object is that everything is done within its scope, that way of anything errors, there is a full rollback. Since functions can call other functions wihtin a transaction, that themselves have a transaction, it's safer to just put everything in the context.